File: src/node/magma/app/compile/type/VariadicType.ts
import { Type } from "../../../../magma/api/Type";
export class VariadicType implements Type {
	type: Type;
	constructor (type: Type) {
		this.type = type;
	}
	generate(): string {
		return this.type.generate() + "[]";
	}
	isFunctional(): boolean {
		return false;
	}
	isVar(): boolean {
		return false;
	}
	generateBeforeName(): string {
		return "...";
	}
}

File: src/java/magma/app/compile/type/VariadicType.java
package magma.app.compile.type;

public record VariadicType(Type type) implements Type {
    @Override
    public String generate() {
        return this.type.generate() + "[]";
    }

    @Override
    public boolean isFunctional() {
        return false;
    }

    @Override
    public boolean isVar() {
        return false;
    }

    @Override
    public String generateBeforeName() {
        return "...";
    }

    @Override
    public String generateSimple() {
        return this.generate();
    }
}
